url8db16c7d62_Arsaell_EdT_tgz-pJ8-DATA_MainJ8/src/DATA/Filler.java

```diff
@@ -0,0 +1,980 @@
+				
+				if (!group.getWeekTable().fieldHappensInDay(field, s.getBegin().getDay()) && !group.getWeekTable().fieldHappensInDay(field, (byte)(s.getBegin().getDay() - 1)) && !group.getWeekTable().fieldHappensInDay(field, (byte)(s.getBegin().getDay() + 1)))	{
+					
+					//ps.println(" #0.5 ");
+					
+					if (this.findClassRoom(field.getType(), s) != null)	{
+						
+						//ps.println("  #1");
+						place = this.findClassRoom(field.getType(), s);
+						slot = s;
+						break;
+					}
+				}
+			}
+			
+			//Si c'était pas possible, on restraint la contrainte au jour même
+			if (place == null || slot == null)	{
+				//ps.println("#2 : " + place + " " + slot + " " + slots.size());
+				for (Slot s : slots)	{
+					//ps.println(" #2.5 : " + s);
+					if (!group.getWeekTable().fieldHappensInDay(field, s.getBegin().getDay()))	{
+						//ps.println("  #2.75");
+						if (this.findClassRoom(field.getType(), s) != null)	{
+							place = this.findClassRoom(field.getType(), s);
+							slot = s;
+							//ps.println("   #3 : " + place + " " + slot);
+							break;
+						}
+					}
+				}
+			}
+			
+			//Et encore une fois, en passant cette contrainte à la trappe.
+			if (place == null || slot == null)	{
+				//ps.println("#4 : " + place + " " + slot + " " + slots.size());
+				for (Slot s : slots)	{
+					//ps.println(" #4.5 " + s);
+					if (this.findClassRoom(field.getType(), s) != null)	{
+						place = this.findClassRoom(field.getType(), s);
+						slot = s;
+						//ps.println("   #5 : " + place + " " + slot);
+						break;
+					}
+				}
+			}
+			
+			
+			if (place == null || slot == null)	{
+				//ps.println("#5 : groupSlots = " + groupSlots + "\nTeachSlots = " + teachSlots);
+				ps.println("\n\n\t/!\\ ### Filler.takeCareOf(Group, Field) couldn't find a classroom !\n" + group + " " + field + " " + duration + " " + group.getTeacher(field) + " " + place + " " + slot + "\n\n");
+				return false;
+			}
+			
+			Lesson res = new Lesson(group.getTeacher(field), group, field, place, slot);
+			
+			boolean failed = false;
+			failed = group.addLesson(res) ? failed : true;
+			failed = group.getTeacher(field).addLesson(res) ? failed : true;
+			failed = place.addLesson(res) ? failed : true;
+			
+			if (failed)	{
+				ps.println("\n\n\t/!\\ ### Filler.takeCareOf(Group, Field) couldn't attribute : " + res + "\n\n");
+				return false;
+			}
+			
+			else 	{
+				left = left.substract(duration);
+				this.done.add(res);
+			}
+		}
+		
+		group.getFieldsDone().put(field, true);
+		
+		return true;
+	}
+
+	/**
+	 * Méthode appelée lorsque l'une des méthodes takeCarOf([...]) renvoie false.
+	 * @param start	L'index dans la liste ordonnée de Constrainables auquel a eu lieu l'erreur
+	 * @param order	la liste ordonnée de Constrainables utilisée par la méthode fill()
+	 * @return l'index dans la liste order auquel la méthode fill() doit reprendre.
+	 */
+	private int handleError(int start, HashMap<Constrainable, Double> order) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.